packages/postgres-database/src/simcore_postgres_database/migration/versions/baf0ee1c37dc_service_runs_credit_transaction_indexes.py

```diff
@@ -0,0 +1,62 @@
+"""service runs/credit_transaction indexes
+
+Revision ID: baf0ee1c37dc
+Revises: 1c069f85d5fd
+Create Date: 2024-04-29 15:42:52.926095+00:00
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "baf0ee1c37dc"
+down_revision = "1c069f85d5fd"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.drop_index(
+        "ix_resource_tracker_credit_transactions_product_name",
+        table_name="resource_tracker_credit_transactions",
+    )
+    op.drop_index(
+        "ix_resource_tracker_credit_transactions_transaction_cla_7bac",
+        table_name="resource_tracker_credit_transactions",
+    )
+    # ### end Alembic commands ###
+
+    # Manually create partial index
+    op.execute(
+        sa.DDL(
+            """
+            CREATE INDEX ix_resource_tracker_credit_transactions_status_running ON resource_tracker_service_runs (service_run_status)
```

Comment:
Q: So this index only gets created when the condition is satisfied. Interesting!

Reply:
Yes! very useful for my use case. Every five minutes I query all running services. Generally, there are not a lot of them. This index will create an "internal hash table" that will search for all running services with O(1). But I do not care about different states, therefore when we update/insert/delete we do not need to modify this index for other states as we do not use them.